### PR TITLE
Page load actions now fire on parameter change as well as on mount

### PR DESCRIPTION
**Pull request by Ghrehh**
## Description
Navigating to the same screen with a different set of params e.g. `/resource/1` -> `/resource/2` was not triggering "on screen load" actions as expected.

We were relying on `svelte-spa-router` to remount the screen component, but the router doesn't remount a component if only the parameters change, as mentioned [here](https://github.com/ItalyPaleAle/svelte-spa-router/issues/148#issuecomment-706716065).

This change fixes the issue by making the function that triggers the "on screen load" actions react to changes in `svelete-spa-router` params instead of relying on the `onMount` lifecycle function.

Addresses: 
- https://github.com/Budibase/budibase/issues/7067

### CODE REVIEW

**Comment by Ghrehh** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/client/src/components/Screen.svelte [hunk 1]
```diff
@@ -15,24 +16,30 @@
...
-    if ($screenStore.activeScreen?.onLoad && !$builderStore.inBuilder) {
-      const actions = enrichButtonActions($screenStore.activeScreen.onLoad, {
-        ...$context,
+    if (screenState.activeScreen?.onLoad && !get(builderStore).inBuilder) {
+      const actions = enrichButtonActions(screenState.activeScreen.onLoad, {
+        ...get(context),
         url: params,
       })
-      if (actions != null) {
+      if (actions !== null) {
```
Comment (aptkingston) [inline packages/client/src/components/Screen.svelte:32]: Let's keep this as `!=` so we account for both `null` and `undefined` values.

**Review by aptkingston — APPROVED** [review]
LGTM! Excellent work @Ghrehh and really impressive to be making your first contribution (and a client library bug no less) so soon!

File: packages/client/src/components/Screen.svelte (hunk 1 above)
Comment (Ghrehh) [reply packages/client/src/components/Screen.svelte]: Ah apologies, missed this, bad copy and paste.

**Review by shogunpurple — APPROVED** [review]
LGTM! First PR in the books 👌

**Review by Rory-Powell — APPROVED** [review]
Nice fix 👍

**Comment by joebudi** [comment]
Awesome work @Ghrehh .

**Pull request merged**